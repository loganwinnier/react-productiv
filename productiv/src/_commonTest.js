const TEST_TODO = {
  id: 1,
  title: "todo",
  description: "desc",
  priority: 1 }

const TEST_TODOS = [
  {
  id: 1,
  title: "todo-1",
  description: "desc-1",
  priority: 1
  },
  {
    id: 2,
    title: "todo-2",
    description: "desc-2",
    priority: 2
  },
  {
    id: 3,
    title: "todo-3",
    description: "desc-3",
    priority: 3
  }]

export { TEST_TODO, TEST_TODOS }
