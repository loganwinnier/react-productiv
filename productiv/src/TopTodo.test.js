import React from "react";
import { fireEvent, render } from "@testing-library/react";
import TopTodo from "./TopTodo";
import { TEST_TODOS } from "./_commonTest";

describe("TopTodo component", function () {
  it("renders without crashing", function () {
    render(<TopTodo todos={TEST_TODOS} />);
  });

  it("contains title, description, priority", function () {
    const result = render(<TopTodo todos={TEST_TODOS} />);
  });

  it("returns null on empty input", function () {
    expect(TopTodo({ todos: [] })).toEqual(null);
  });

  it("renders correct top todo", function () {
    const result = render(<TopTodo todos={TEST_TODOS} />);
    expect(result.queryByText(TEST_TODOS[0].title)).toBeInTheDocument();
    expect(result.queryByText(TEST_TODOS[0].description)).toBeInTheDocument();
    expect(result.queryByText(TEST_TODOS[0].priority)).toBeInTheDocument();

  });

  it("matches snapshot", function () {
    const { container } = render(<TopTodo todos={TEST_TODOS} />);
    expect(container).toMatchSnapshot();
  });
});