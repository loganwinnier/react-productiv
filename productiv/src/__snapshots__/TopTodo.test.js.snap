// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopTodo component matches snapshot 1`] = `
<div>
  <div
    class="Todo"
  >
    <div>
      <b>
        todo-1
      </b>
       
      <small>
        1
      </small>
    </div>
    <div>
      <small>
        desc-1
      </small>
    </div>
  </div>
</div>
`;
